ethereum/blockwatch/block_watcher.go

```diff
@@ -302,14 +316,18 @@ func (w *Watcher) SyncToLatestBlock() error {
 		return syncErr
 	}
 	if w.shouldRevertChanges(lastStoredHeader, allEvents) {
```

Comment:
This function returns false if `len(events) == 0 || lastStoredHeader == nil`, but `len(events)` can't be since we would have returned in line 301 if so. I think that if `lastStoredHeader` is `nil` in line 242 (old) /  256 (new), then it will still be nil here, so maybe we should just do the revert there?

Reply:
`lastStoredHeader` is updated on line 281 (of the version of the code that I'm looking at).

Reply:
This logic doesn't make sense to me. If the function returns false, then we don't want to revert the changes. 

As a follow up, we don't know `nextLatestHeader` until after the for-loop (because `events` is added to by `buildCanonicalChain`). Having said this, I don't think that we can do the revert earlier unless me make a different refactor first.

Reply:
No worries, this happens to me all the time. I'll check out the deadcode. If the event length is 0, then we really don't want to revert changes, so I want to be careful that this change wouldn't introduce problems if the function was re-used elsewhere.

Reply:
I think I'm going to leave this in for now. Let's address this in a refactor, which may entail completely eliminating this function.

My rationale is that removing the deadcode makes the function more context specific (you have to check `len(events) == 0` elsewhere for it to work properly). This is okay in this case because it's only called once, but I don't really like that since we are encapsulating logic in a function that should be reusable.